### PR TITLE
[DOCS] NPU properties in documentation for master

### PR DESCRIPTION
**Pull request by msmykx-intel**
Port from https://github.com/openvinotoolkit/openvino/pull/25590

* adding newly supported properties to npu-device documentation
* addig section describing compilation-mode-params to npu-device documentation

### CODE REVIEW

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst [hunk 1]
```diff
@@ -155,6 +162,60 @@ offer a limited set of supported OpenVINO features.
    based on the performance mode is **4 for THROUGHPUT** and **1 for LATENCY**.
    The default mode for the NPU device is LATENCY.
 
+**ov::intel_npu::compilation_mode_params**
+
+``ov::intel_npu::compilation_mode_params`` is an NPU-specific property that allows to
+control model compilation for NPU.
```
Comment (sgolebiewski-intel) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst:168]:
```suggestion
``ov::intel_npu::compilation_mode_params`` is an NPU-specific property that allows
control of model compilation for NPU.
```

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst [hunk 2]
```diff
@@ -155,6 +162,60 @@ offer a limited set of supported OpenVINO features.
...
+
+.. note::
+
+   The functionality is in experimental stage currently, can be a subject for
+   deprecation and may be replaced with generic OV API in future OV releases.
```
Comment (sgolebiewski-intel) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst:173]:
```suggestion
   The functionality is in an experimental stage currently, can be a subject for
   deprecation and may be replaced with a generic OV API in future OV releases.
```

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst [hunk 3]
```diff
@@ -155,6 +162,60 @@ offer a limited set of supported OpenVINO features.
...
+
+Following configuration options are supported:
```
Comment (sgolebiewski-intel) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst:175]:
```suggestion
The following configuration options are supported:
```

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst [hunk 4]
```diff
@@ -155,6 +162,60 @@ offer a limited set of supported OpenVINO features.
...
+Defines a preset of optimization passes to be applied during compilation.
+
+.. list-table::
+   :widths: 10 200
+   :header-rows: 1
+
+   * - **Value**
+     - **Description**
+   * - 0
+     - Reduced subset of optimization passes. Smaller compile time.
```
Comment (sgolebiewski-intel) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst:188]:
```suggestion
     - A reduced subset of optimization passes. Smaller compile time.
```

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst [hunk 5]
```diff
@@ -155,6 +162,60 @@ offer a limited set of supported OpenVINO features.
...
+   * - 1
+     - **Default.** Balanced performance/compile time.
+   * - 2
+     - Prioritize performance over compile time that may be an issue.
+
+**performance-hint-override**
+
+An extension for LATENCY mode being specified using ``ov::hint::performance_mode``
+Has no effect for other ``ov::hint::PerformanceMode`` hints.
```
Comment (sgolebiewski-intel) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst:197]:
```suggestion
It has no effect on the other ``ov::hint::PerformanceMode`` hints.
```

File: docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst [hunk 6]
```diff
@@ -155,6 +162,60 @@ offer a limited set of supported OpenVINO features.
...
+   * - 0
+     - Reduced subset of optimization passes. Smaller compile time.
+   * - 1
+     - **Default.** Balanced performance/compile time.
+   * - 2
+     - Prioritize performance over compile time that may be an issue.
+
+**performance-hint-override**
+
+An extension for LATENCY mode being specified using ``ov::hint::performance_mode``
```
Comment (sgolebiewski-intel) [inline docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/npu-device.rst:196]:
```suggestion
An extension for LATENCY mode being specified using ``ov::hint::performance_mode``.
```

**Comment by msmykx-intel** [comment]
Further adjustments will be made in a separate PR.

**Pull request merged**